### PR TITLE
Improve error messaging for invalid arguments

### PR DESCRIPTION
**Pull request by pingsutw**
## Tracking issue
NA

## Why are the changes needed?
When users encounter this error, they do not know what to do

## What changes were proposed in this pull request?
Improve error messaging for invalid arguments.

## How was this patch tested?
pyflyte register 

### Screenshots
Before:
Screenshot 2024-08-12 at 4 03 49 PM

After:
Screenshot 2024-08-12 at 3 53 01 PM

## Check all the applicable boxes 

- [x] I updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Related PRs
NA

## Docs link
NA

### CODE REVIEW

**Comment by austin362667** [comment]
https://github.com/flyteorg/flyte/blob/c5168d1691969ef1c680dc9dec8ee00d91af5d08/flyteadmin/pkg/errors/errors_test.go#L77

The assertion in `TestJsonDifferHasDiffError` should modify `\t\t- /four: 4 -> 0` to `/four:\n\t- 4 \n\t+ 0` as well..

File: flyteadmin/pkg/errors/errors.go [hunk 1]
```diff
@@ -112,31 +112,30 @@ func NewIncompatibleClusterError(ctx context.Context, errorMsg, curCluster strin
...
+			result := fmt.Sprintf("%v:\n\t- %v \n\t+ %v", obj.Path, obj.Value, val.Value)
 			results = append(results, result)
 		}
 	}
 	return results
 }
 
 func NewTaskExistsDifferentStructureError(ctx context.Context, request *admin.TaskCreateRequest, oldSpec *core.CompiledTask, newSpec *core.CompiledTask) FlyteAdminError {
-	errorMsg := fmt.Sprintf("%v task with different structure already exists:\n", request.Id.Name)
+	errorMsg := fmt.Sprintf("%v task with different structure already exists. (Please register a new version of the task):\n", request.Id.Name)
```
Comment (austin362667) [inline flyteadmin/pkg/errors/errors.go:131]: This actionable hint is nice!

File: flyteadmin/pkg/errors/errors.go [hunk 2]
```diff
@@ -112,31 +112,30 @@ func NewIncompatibleClusterError(ctx context.Context, errorMsg, curCluster strin
 }
 
 func compareJsons(jsonArray1 jsondiff.Patch, jsonArray2 jsondiff.Patch) []string {
-	results := []string{}
+	var results []string
```
Comment (austin362667) [inline flyteadmin/pkg/errors/errors.go:115]: Should add a nil check if initializing a zero value(which is nil) slice is intended?

File: flyteadmin/pkg/errors/errors_test.go [hunk 3]
```diff
@@ -284,7 +284,7 @@ func TestNewLaunchPlanExistsDifferentStructureError(t *testing.T) {
 	s, ok := status.FromError(statusErr)
 	assert.True(t, ok)
 	assert.Equal(t, codes.InvalidArgument, s.Code())
-	assert.Equal(t, "launch plan with different structure already exists:\n\t\t- /workflow_id/version: ver1 -> ver2", s.Message())
+	assert.Equal(t, "t1 launch plan with different structure already exists. (Please register a new version of the launch plan):\n/workflow_id/version:\n\t- ver1 \n\t+ ver2", s.Message())
```
Comment (austin362667) [inline flyteadmin/pkg/errors/errors_test.go:287]:
```suggestion
	assert.Equal(t, "hello launch plan with different structure already exists. (Please register a new version of the launch plan):\n/workflow_id/version:\n\t- ver1 \n\t+ ver2", s.Message())
```

**Review by austin362667 — APPROVED** [review]

**Review by austin362667 — APPROVED** [review]

File: flyteadmin/pkg/errors/errors_test.go (hunk 3 above)
Comment (eapolinario) [inline flyteadmin/pkg/errors/errors_test.go:287]: Can we have a different identifier (of the [launcplan type](https://github.com/flyteorg/flyte/blob/37b4e13ac4a3594ac63b7a35058f4b2220e51282/flyteadmin/pkg/common/entity.go#L28)) in this test?

**Review by eapolinario — APPROVED** [review]

**Pull request merged**